src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/VMDynamicTests.ps1

```diff
@@ -0,0 +1,341 @@
+    $vnet = Get-AzureVirtualNetwork -Name ('vnet' + $rgname) -ResourceGroupName $rgname;
+    $subnetId = $vnet.Subnets[0].Id;
+
+'@;
+
+    $min_num_of_nic_ids = 1;
+    $max_num_of_nic_ids = 1; # TODO: Number of NICs is related to VM Size, and no API to get that information yet.
+    $num_of_nic_ids = Get-Random -Minimum $min_num_of_nic_ids -Maximum (1 + $max_num_of_nic_ids) -SetSeed $random_seed;
+    
+    $fn_body =
+@"
+    `$nic_ids = @(`$null) * ${num_of_nic_ids};
+"@;
+
+    $primary_id = Get-Random -Minimum 0 -Maximum $max_num_of_nic_ids -SetSeed $random_seed;
+
+    for ($i = 0; $i -lt $num_of_nic_ids; $i++)
+    {
+        $nic_var_name = '$nic' + $i;
+        $nic_name_str = "('nic${i}'" + " + `$rgname)";
+        $primary_switch_text = '';
+        if (($num_of_nic_ids -gt 1) -and ($i -eq $primary_id))
+        {
+            $primary_switch_text = ' -Primary';
+        }
+        elseif ($num_of_nic_ids -eq 1)
+        {
+            $primary_switch_text = (' -Primary', ' ') | Get-Random -SetSeed $random_seed;
+            $primary_switch_text = $primary_switch_text.TrimEnd();
+        }
+
+        $fn_body +=
+@"
+
+    ${nic_var_name} = New-AzureNetworkInterface -Force -Name ${nic_name_str} -ResourceGroupName `$rgname -Location `$loc -SubnetId `$subnetId;
+    `$nic_ids[$i] = ${nic_var_name}.Id;
+    `$vmconfig = Add-AzureVMNetworkInterface -VM `$vmconfig -Id ${nic_var_name}.Id${primary_switch_text};
+
+"@;
+    }
+
+
+$fn_end =
+@'
+
+    return $nic_ids;
+}
+'@;
+
+    return $fn_start + $fn_body + $fn_end;
+}
+
+
+$func_create_and_setup_vm_config_object =
+@'
+
+function create_and_setup_vm_config_object
+{
+    param ([string] $loc, [string] $rgname, [string] $vmsize)
+
+    $vmconfig = get_vm_config_object $rgname $vmsize
+
+    $user = "Foo12";
+    $password = "BaR#123" + $rgname;
+    $securePassword = ConvertTo-SecureString $password -AsPlainText -Force;
+    $cred = New-Object System.Management.Automation.PSCredential ($user, $securePassword);
+    $computerName = "cn" + $rgname;
+    $vmconfig = Set-AzureVMOperatingSystem -VM $vmconfig -Windows -ComputerName $computerName -Credential $cred;
+
+    return $vmconfig;
+}
+
+'@;
+
+
+
+$func_setup_image_and_disks =
+@'
+
+function setup_image_and_disks
+{
+    param ([string] $loc, [string] $rgname, [string] $stoname, $vmconfig)
+
+    $osDiskName = 'osDisk';
+    $osDiskVhdUri = "https://$stoname.blob.core.windows.net/test/os.vhd";
+    $osDiskCaching = 'ReadWrite';
+
+    $vmconfig = Set-AzureVMOSDisk -VM $vmconfig -Name $osDiskName -VhdUri $osDiskVhdUri -Caching $osDiskCaching -CreateOption FromImage;
+
+    # Image Reference;
+    $vmconfig.StorageProfile.SourceImage = $null;
+    $imgRef = Get-DefaultCRPImage;
+    $vmconfig = ($imgRef | Set-AzureVMSourceImage -VM $vmconfig);
+
+    # TODO: Remove Data Disks for now
```

Comment:
Is it really 'TODO'?  Data disks is actually removed by the following line, so I think TODO is misleading.  Can you add more detailed comments why data disks are removed and what is to do (e.g. a bug number?).